File: src/gestorAplicacion/restaurante/Caja.java
package gestorAplicacion.restaurante;
import java.util.ArrayList; 
import java.util.Collections;
import java.io.Serializable;
public class Caja implements Serializable{
	private int efectivo;
	public ArrayList<Integer> ingresos;
	private ArrayList<Integer>  egresos;    
	static private ArrayList<Integer>  totalingresos = new ArrayList<Integer>();
	static private ArrayList<Integer>  totalegresos = new ArrayList<Integer>();
	
	
	//contructores
	public Caja() {
		this(0,new  ArrayList<Integer>(),new  ArrayList<Integer>());
	}
	public Caja(int efectivo, ArrayList<Integer> ingresos, ArrayList<Integer> egresos) {
		this.efectivo = efectivo;
		this.ingresos = ingresos;
		this.egresos = egresos;	
	}

	
	
	//metodos get
	public int getEfectivo() {
		return efectivo;
	}
	public ArrayList<Integer> getIngresos(){
		return ingresos;
	}
	public ArrayList<Integer> getEgresos(){
		return egresos;
	}
	public static ArrayList<Integer> getTotalIngresos(){
		return totalingresos;
	}
	
	public static ArrayList<Integer> getTotalEgresos(){
		return totalegresos;
	}

	
	
	//metodos set 
	public void setEfectivo(int n) {
		efectivo = n;
	}

	public void nuevoIngreso(int n) {
		ingresos.add(n);
		totalingresos.add(n);
	}
	public void nuevoEgreso(int n) {
		egresos.add(n);
		totalegresos.add(n);
	}
	
	
	//metodos reales
	public String devuelta(int n, Orden orden) {
		if (n > orden.getPrecio_total()) {
			int w = n - orden.getPrecio_total();   //devuelta 
			this.nuevoIngreso(orden.getPrecio_total());     //agrego ingresos a la lista
			return  w + "";                    // convierto el int a string 
		}
		else {
			return "Cantidad Insuficiente";
		}
	}
	
	public int arqueo() {
		int x = 0;
		for (int n: ingresos) {
			x += n;
		}
		return x;
	}
	
	public int arqueoTotal() {
		int x = 0;
		for (int n: totalingresos) {
			x += n;
		}
		return x;
	}
}

File: Python/gestorAplicacion/restaurante/caja.py
class Caja:

    #constructor
    def __init__(self, efectivo = 0, ingresos = None, egresos = None):
        if ingresos is None:
            ingresos = []
        if egresos is None:
            egresos = []
        self._efectivo = efectivo
        self._ingresos = ingresos
        self._egresos = egresos

    #metodos get 
    def getEfectivo(self):
        return self._efectivo

    def getIngresos(self):
        return self._ingresos

    def getEgresos(self):
        return self._egresos

    #metodos set
    def setEfectivo(self, efectivo):
        self._efectivo += efectivo

    def setIngresos(self, ingresos):
        self._ingresos.append(ingresos)

    def setEgresos(self, egresos):
        self._egresos.append(egresos)

    #metodos
    def devuelta(self, n, orden):
        if n > orden.getPrecio_total():
            w = n - orden.getPrecioTotal()
            return w + ""
        else:
            return "cantidad insuficiente"

    def arqueo(self):
        x = 0
        for i in self._ingresos:
            x += i
        for i in self._egresos:
            x -= i
        return x